test/test_sparse.py

```diff
@@ -1976,10 +1986,15 @@ def _test_sparse_mask_hybrid_fixed():
                 [[13, 5], [14, 1], [15, 1], [16, 6]],
                 [[17, 7], [18, 2], [19, 7], [20, 1]],
             ])
-            res = dense.sparse_mask(x)
+            res_dense_lhs = dense.sparse_mask(x)
+            sparse = dense.to_sparse(2)
+            res_sparse_lhs = sparse.sparse_mask(x)
```

Comment:
In this test, we have `sparse.sparse_dim() == x.sparse_dim()`. Could you add a test where `sparse.sparse_dim() != x.sparse_dim()` so that `restrict_indices_to_rhs` is tested? Say:
```
sparse3 = dense.to_sparse(3)
res_sparse_lhs3 = sparse3.sparse_mask(x)
self.assertEqual(res_dense_lhs3, expected, exact_is_coalesced=False)
```

Reply:
This will not work, the intersection kernel assumes the same number of and equal sparse dims.

Reply:
OK, sounds like an interesting follow-up.

Reply:
It is a part of a different kernel, similar to what I did for the dense-sparse intersection. It is always possible to bring them down to equal sparse dim and reuse this kernel.